### PR TITLE
Store retry attempts in example metadata.

### PR DESCRIPTION
**Pull request by aeberlin**
Simple shim to store attempted retries in the example metadata.

Purpose is to be able to reference this count in custom reporters or summary loggers.

Thanks, cheers!

### CODE REVIEW

**Comment by michaelglass** [comment]
please add a test confirming this functionality works, thanks!

**Comment by aeberlin** [comment]
@michaelglass I've added a test, but oddly it's failing on 3.2 builds. Do you have any ideas why?

**Comment by michaelglass** [comment]
sorry, I can't. If we drop 3.2 support, will consider merging this.

**Comment by aeberlin** [comment]
Hi @michaelglass,

After taking a fresh look earlier this morning, I realized that the way I was instantiating a new example group (and example) in the RSpec DSL would not work the same way for versions < 3.3.

I have rebased my branch off of current master, and made changes to the test in order to reflect the difference. I wish less meta was involved, but it was the only way I could be absolutely sure that all the versions are supported.

Please let me know if there's anything else I can do to get this merged.

**Comment by aeberlin** [comment]
@michaelglass Could I please get some feedback here?

Thanks!

File: lib/rspec/retry.rb [hunk 1]
```diff
@@ -94,6 +94,8 @@ def run
           end
         end
 
+        example.metadata[:retry_attempts] = self.attempts
+        # require 'pry'; binding.pry
```
Comment (dwbutler) [inline lib/rspec/retry.rb]: Can you please remove this debugging code?

**Comment by dwbutler** [comment]
It looks fine to me, though the spec is pretty gnarly. Kind of makes you wish RSpec were easier to test with RSpec.

**Comment by michaelglass** [comment]
I think I want to do a major version release when we include this and drop rspec 3 support. It smells weird for some features to be rspec version dependent

File: lib/rspec/retry.rb (hunk 1 above)
Comment (aeberlin) [inline lib/rspec/retry.rb]: Done, sorry about that.

**Comment by aeberlin** [comment]
I truly wish testing RSpec with RSpec was easier, but I doubt it will ever be so.

That sounds good to me, thanks for the feedback!

**Comment by michaelglass** [comment]
moved to #97

**Pull request closed without merging**